internal/xds/util.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ClusterLoadAssignmentName generates the name used for an EDS
+// ClusterLoadAssignment, given a fully qualified Service name and
+// port. This name is a contract between the producer of a cluster
+// (i.e. the EDS service) and the consumer of a cluster (most likely
+// a HTTP Route Action).
+func ClusterLoadAssignmentName(service types.NamespacedName, portName string) string {
```

Comment:
Should this be in the `internal/envoy` package?

Reply:
It was previously in the envoy package, but now that it is used in the `dag` package, that resulted in import loops that I needed to break.

Reply:
oh I see now, makes sense. Unfortunate though this is getting all mixed around.